### PR TITLE
Add Jsonnet language.

### PR DESCRIPTION
**Pull request by sparkprime**
This is an update of the following PR which has now been accepted.  

https://github.com/github/linguist/pull/2653

@pchaigno 

Things that I updated:
- Update submodule to latest language-jsonnet
- Add libsonnet and libjsonnet extensions
- Change colour to the one used by the new website and logo

### CODE REVIEW

**Comment by sparkprime** [comment]
I ran harvester on .libsonnet and it returned 244 repos with
`xsel -b | cut -f 5 -d '/' | sort -u | wc`

The .libjsonnet extension is much less used, we can drop it if it's a problem

**Comment by lildude** [comment]
> The .libjsonnet extension is much less used, we can drop it if it's a problem

Yes, please. A simple GitHub search shows only 10 uses - https://github.com/search?q=extension%3Alibjsonnet+NOT+nothack&type=Code

`.libsonnet` appears sufficiently widespread to include support.

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -2423,6 +2423,16 @@ Jolie:
   ace_mode: text
   tm_scope: source.jolie
   language_id: 998078858
+Jsonnet:
+  color: "#0064bd"
+  type: programming
+  ace_mode: text
+  extensions:
+  - ".jsonnet"
+  - ".libjsonnet"
```
Comment (lildude) [inline lib/linguist/languages.yml]: 🔥 this extension for now - it's no where near prevalent enough at only 10 files.

Comment (sparkprime) [reply lib/linguist/languages.yml]: Done.

**Comment by lildude** [comment]
@pchaigno or @Alhadis wanna take a quick peek at this in case I'm missing anything. I'm in the process of testing a new release and I think we can squeeze this one in.

**Comment by Alhadis** [comment]
LGTM. :+1:

**Pull request merged**